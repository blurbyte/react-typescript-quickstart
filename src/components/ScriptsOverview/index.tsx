import * as React from 'react';

import SecondaryHeadline from '../SecondaryHeadline';
import Text from '../Text';
import Table from '../Table';

class ScriptsOverview extends React.Component {
  shouldComponentUpdate() {
    return false;
  }

  render() {
    return (
      <section>
        <SecondaryHeadline>NPM Scripts overview</SecondaryHeadline>
        <Text>Just a grasp of what could be found in package.json:</Text>
        <Table>
          <thead>
            <tr>
              <Table.HeadCell>Script name</Table.HeadCell>
              <Table.HeadCell>Description</Table.HeadCell>
            </tr>
          </thead>
          <tbody>
            <Table.Row>
              <Table.Cell>start</Table.Cell>
              <Table.Cell>Launches app in development mode</Table.Cell>
            </Table.Row>
            <Table.Row>
              <Table.Cell>build</Table.Cell>
              <Table.Cell>Creates production, ready to deploy build of the app</Table.Cell>
            </Table.Row>
            <Table.Row>
              <Table.Cell>test</Table.Cell>
              <Table.Cell>Runs whole test suite</Table.Cell>
            </Table.Row>
            <Table.Row>
              <Table.Cell>test:watch</Table.Cell>
              <Table.Cell>Watch files for changes and rerun related tests</Table.Cell>
            </Table.Row>
            <Table.Row>
              <Table.Cell>test:update</Table.Cell>
              <Table.Cell>Updates all test snapshots</Table.Cell>
            </Table.Row>
            <Table.Row>
              <Table.Cell>lint</Table.Cell>
              <Table.Cell>Checks the whole project for linting errors</Table.Cell>
            </Table.Row>
            <Table.Row>
              <Table.Cell>typecheck</Table.Cell>
              <Table.Cell>Runs TypeScript compiler to find type errors</Table.Cell>
            </Table.Row>
            <Table.Row>
              <Table.Cell>format</Table.Cell>
              <Table.Cell>Formats code with prettier</Table.Cell>
            </Table.Row>
          </tbody>
        </Table>
      </section>
    );
  }
}

export default ScriptsOverview;
