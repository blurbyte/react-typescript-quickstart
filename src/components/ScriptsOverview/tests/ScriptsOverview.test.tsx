import * as React from 'react';
import { shallow } from 'enzyme';

import ScriptsOverview from '../index';

describe('ScriptsOverview', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<ScriptsOverview />);
    expect(wrapper).toMatchSnapshot();
  });
});
