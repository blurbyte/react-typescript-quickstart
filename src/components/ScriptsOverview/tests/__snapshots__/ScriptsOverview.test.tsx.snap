// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScriptsOverview renders correctly 1`] = `
<section>
  <styled.h2>
    NPM Scripts overview
  </styled.h2>
  <styled.p>
    Just a grasp of what could be found in package.json:
  </styled.p>
  <Table>
    <thead>
      <tr>
        <Styled(styled.th)>
          Script name
        </Styled(styled.th)>
        <Styled(styled.th)>
          Description
        </Styled(styled.th)>
      </tr>
    </thead>
    <tbody>
      <styled.tr>
        <styled.td>
          start
        </styled.td>
        <styled.td>
          Launches app in development mode
        </styled.td>
      </styled.tr>
      <styled.tr>
        <styled.td>
          build
        </styled.td>
        <styled.td>
          Creates production, ready to deploy build of the app
        </styled.td>
      </styled.tr>
      <styled.tr>
        <styled.td>
          test
        </styled.td>
        <styled.td>
          Runs whole test suite
        </styled.td>
      </styled.tr>
      <styled.tr>
        <styled.td>
          test:watch
        </styled.td>
        <styled.td>
          Watch files for changes and rerun related tests
        </styled.td>
      </styled.tr>
      <styled.tr>
        <styled.td>
          test:update
        </styled.td>
        <styled.td>
          Updates all test snapshots
        </styled.td>
      </styled.tr>
      <styled.tr>
        <styled.td>
          lint
        </styled.td>
        <styled.td>
          Checks the whole project for linting errors
        </styled.td>
      </styled.tr>
      <styled.tr>
        <styled.td>
          typecheck
        </styled.td>
        <styled.td>
          Runs TypeScript compiler to find type errors
        </styled.td>
      </styled.tr>
      <styled.tr>
        <styled.td>
          format
        </styled.td>
        <styled.td>
          Formats code with prettier
        </styled.td>
      </styled.tr>
    </tbody>
  </Table>
</section>
`;
